pkg/pgmodel/ingestor/ingestor.go

```diff
@@ -74,34 +140,46 @@ func (ingestor *DBIngestor) Ingest(r *prompb.WriteRequest) (uint64, error) {
 			return 0, errors.ErrNoMetricName
 		}
 		sample := model.NewPromSample(seriesLabels, ts.Samples)
-		totalRows += uint64(len(ts.Samples))
+		totalSamplesRows += uint64(len(ts.Samples))
 
 		dataSamples[metricName] = append(dataSamples[metricName], sample)
 		// we're going to free req after this, but we still need the samples,
 		// so nil the field
 		ts.Samples = nil
 	}
-	// WriteRequests can contain pointers into the original buffer we deserialized
-	// them out of, and can be quite large in and of themselves. In order to prevent
-	// memory blowup, and to allow faster deserializing, we recycle the WriteRequest
-	// here, allowing it to be either garbage collected or reused for a new request.
-	// In order for this to work correctly, any data we wish to keep using (e.g.
-	// samples) must no longer be reachable from req.
-	FinishWriteRequest(r)
```

Comment:
Memory needs to be cleared here, since the data is no longer used. Check previous comment for a suggestion.